src/main/java/task/TaskList.java

```diff
@@ -0,0 +1,143 @@
+                    "\nNow you have " + tasks.size() + " tasks in the list.");
+        }
+    }
+
+    /**
+     * Marks a specific task in the task list as completed.
+     * @param number String number that specifies task number in task list to mark as completed.
+     */
+    public void mark(String number) {
+        int num = Integer.parseInt(number);
+        tasks.get(num - 1).setAsDone();
+        Task temp = tasks.get(num - 1);
+        System.out.println("Nice! :P I've marked this task as done:\n [" + temp.symbol() + "][" +
+                temp.getStatusIcon() + "] " + temp);
+    }
+
+    /**
+     * Marks a specific task in the task list as not completed.
+     * @param number String number that specifies task number in task list to mark as not completed.
+     */
+    public void unmark(String number) {
+        int num = Integer.parseInt(number);
+        tasks.get(num - 1).setAsNotDone();
+        Task temp = tasks.get(num - 1);
+        System.out.println("OK ._. , I've marked this task as not done yet:\n [" + temp.symbol() + "][" +
+                temp.getStatusIcon() + "] " + temp);
+    }
+
+    /**
+     * Deletes all or specific tasks from the task list based on string input.
+     * @param number string that specifies if a specific task number or all tasks are to be deleted.
+     */
+    public void delete(String number) {
+        if (number.equals("all")) {
+            this.tasks =  new ArrayList<>();
+            System.out.println("All right! I have deleted all tasks in your list.");
+        } else {
+            int num = Integer.parseInt(number);
+            Task temp = tasks.get(num - 1);
+            System.out.println("Noted. I've removed this task:\n [" + temp.symbol() + "][" +
+                    temp.getStatusIcon() + "] " + temp +
+                    "\nNow you have " + (tasks.size() - 1) + " tasks left in this list");
+            tasks.remove(num - 1);
+        }
+    }
+
+    /**
+     * Prints tasks in the task list that contain a specified keyword.
+     * @param inputStringsArray keyword used to filer tasks in task list.
+     */
+    public void find(String inputStringsArray) {
+        ArrayList<Task> foundTasks = new ArrayList<>();
+        for (int i = 0; i < this.tasks.size(); i++) {
+            String details = this.tasks.get(i).details;
+            String[] detailsArray = details.split(" ");
+            for (int j = 0; j < detailsArray.length; j++) {
+                if (detailsArray[j].equals(inputStringsArray)) {
+                    foundTasks.add(this.tasks.get(i));
+                    break;
+                }
+            }
+        }
+        String returnString = "Here are the matching tasks in your list:\n";
+        for (int i = 0; i < foundTasks.size(); i++) {
+            returnString = returnString + (i + 1) + ". [" + foundTasks.get(i).symbol() + "]["
+                    + foundTasks.get(i).getStatusIcon() + "] " + foundTasks.get(i).displayTime() + "\n";
```

Comment:
Line break for long lines would make the code look neater 👍  (e.g. after operator "+")